test/interval_tests/construction.jl

```diff
@@ -310,3 +310,10 @@ end
 @testset "a..b with a > b" begin
     @test_throws ArgumentError 3..2
 end
+
+@testset "Hashing of Intervals" begin
+    x = Interval{Float64}(1, 2)
+    y = Interval{BigFloat}(1, 2)
+    @test isequal(x, y)
+    @test isequal(hash(x), hash(y))
```

Comment:
Using your definitions of `x` and `y`, `x == y` is `true` in current master, but `x === y` is `false`. Is it ok in that case that `isequal(hash(x), hash(y))` yields `true`?

Reply:
According to the definition of the `hash` function, I think this should be fine.
```
help?> hash
  hash(x[, h::UInt])

  Compute an integer hash code such that isequal(x,y) implies
  hash(x)==hash(y). The optional second argument h is a hash code to be mixed
  with the result.
```

Reply:
I've tried your implementation with other less trivial cases, which I think you should include as tests. For instance:
`x = @interval(0.1)` and `y = IntervalArithmetic.big53(x)`. In master `x == y` yields `true`, and with your PR we get that `hash(x) == hash(y)` is `true` too. Probably including also a case resulting `false` in an obvious way should also be there.

Reply:
Alright, I will. Thanks!

Reply:
And add some docstrings or some explanation in the documentation :-)